File: src/main/java/com/mbsl/controller/ReportGenerate.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mbsl.controller;

import com.mbsl.model.User;
import com.mbsl.service.DonloadRepoService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RequestBody;

/**
 *
 * @author MBSL2523
 */
public class ReportGenerate implements IReportGenerate {

    @Autowired
    private DonloadRepoService donloadRepoService;

    @Override
    public ResponseEntity<User> donloadRepo(@RequestBody User repo) {

        return new ResponseEntity<>(donloadRepoService.donloadRepo(repo), HttpStatus.OK);

    }

}

File: src/main/resources/static/js/app/reportGenerate.js
/* 
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/JSP_Servlet/JavaScript.js to edit this template
 */

app.controller("reportGenerate", function ($scope, $http, $mdDialog, $filter) {

    $scope.clicks =["Available Users", "Transfer Users" ,"Inactivation Users" ,"temporary  Access"];

    $scope.repo = {

        rqstId: "",
        epfNo: "",
        userName: "",
        fullName: "",
        designation: "",
        emailAddress: "",
        branchorDept: "",
        profile: "",
        password: "",
        mobileNo: "",
        click: "",
        status: "",
        currentTill: "",
        click2: "",
        newBranchOrDept: "",
        newProfile: "",
        newDesignation: "",
        newTill: "",
        reason: "",
        startDate: "",
        endDate: "",
        rqstType: ""


    };

    $scope.repo.toDay = $filter('date')(new Date(), 'MM/dd/yyyy');


    $scope.donloadRepo = function () {


        var method = "POST";
        var url = 'report/dwn_report';
        $scope.createUserBusy = $http({
            method: method,
            url: url,
            data: angular.toJson($scope.repo),
            headers: {
                'Content-Type': 'application/json'
            },
            transformResponse: [
                function (data) {

                    console.log($scope.repo);
                }
            ]
        });

    };

});
